### PR TITLE
bài internet connection

### PR DESCRIPTION
**Pull request by Ngduong29**
## Related Tickets
- [#Task xxxxx: Content](https://edu-redmine.sun-asterisk.vn/issues/xxxx)
## WHAT
-  bài slide internet connection
## Evidence (Screenshot or Video)
image
image

## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/blob/master/eng/android/coding_convention_android_kotlin.md | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(Other notes)*

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -91,4 +97,4 @@
             android:exported="false" />
     </application>
 
-</manifest>
+</manifest>
```
Comment (hungvv-1301) [inline app/src/main/AndroidManifest.xml]:
```suggestion
</manifest>

```

**Review by minhnn-1279 — APPROVED** [review]

File: app/src/main/java/com/sun/android/internet_connection/WhoWroteItActivity.kt [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                }
+            }
+        }
+    }
+
+    override fun onCreateLoader(id: Int, args: Bundle?): Loader<String> {
+        var queryString = ""
+
+        if (args != null) {
+            queryString = args.getString("queryString").toString()
```
Comment (hungvv-1301) [inline app/src/main/java/com/sun/android/internet_connection/WhoWroteItActivity.kt]: extract constant

File: app/src/main/java/com/sun/android/internet_connection/WhoWroteItActivity.kt [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            var authors: String? = null
+
+            while (i < itemsArray.length() &&
+                authors == null && title == null
+            ) {
+                val book = itemsArray.getJSONObject(i)
+                val volumeInfo = book.getJSONObject("volumeInfo")
+                try {
+                    title = volumeInfo.getString("title")
+                    authors = volumeInfo.getString("authors")
```
Comment (hungvv-1301) [inline app/src/main/java/com/sun/android/internet_connection/WhoWroteItActivity.kt]: extract constant

File: app/src/main/java/com/sun/android/internet_connection/WhoWroteItActivity.kt [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+                inputManager.hideSoftInputFromWindow(binding.editTextBookInput.windowToken,InputMethodManager.HIDE_NOT_ALWAYS)
+            }
+            val connMgr : ConnectivityManager = getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+            var networkInfo: NetworkInfo? = null
+            if (connMgr != null){
+                networkInfo = connMgr.activeNetworkInfo
+            }
+            if(networkInfo != null && networkInfo.isConnected && queryString.length != 0){
+                val queryBundle = Bundle()
+                queryBundle?.putString("queryString",queryString)
```
Comment (hungvv-1301) [inline app/src/main/java/com/sun/android/internet_connection/WhoWroteItActivity.kt]: .

File: app/src/main/res/layout/activity_who_wrote_it.xml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        android:inputType="text"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/text_instruction"/>
+
+    <Button
+        android:id="@+id/button_search"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="8dp"
```
Comment (hungvv-1301) [inline app/src/main/res/layout/activity_who_wrote_it.xml]: dimens

**Pull request merged**